plugins/woocommerce/src/Internal/DataStores/Orders/OrdersTableDataStore.php

```diff
@@ -2621,8 +2621,15 @@ public function update( &$order ) {
 		$order->apply_changes();
 		$this->clear_caches( $order );
 
-		// For backwards compatibility, moving a draft order to a valid status triggers the 'woocommerce_new_order' hook.
-		if ( ! empty( $changes['status'] ) && in_array( $previous_status, array( 'new', 'auto-draft', 'draft', 'checkout-draft' ), true ) ) {
+		$nonTriggeringOrderStatuses = array( 'new', 'auto-draft', 'draft', 'checkout-draft', 'refunded', 'failed', 'cancelled');
+
+		// For backwards compatibility, this hook should be fired only if the new status is not one of the non-triggering statuses and the previous status was one of the non-triggering statuses.
+		if (
+			! empty( $changes['status'] )
+			&& $changes['status'] !== $previous_status
+			&& ! in_array( $changes['status'], $nonTriggeringOrderStatuses, true )
+			&& in_array( $previous_status, $nonTriggeringOrderStatuses, true )
```

Comment:
Looking at the code, in this context, a brand new unsaved order would have $previous_status set to `status`?

```
$previous_status = ArrayUtil::get_value_or_default( $order->get_data(), 'status' );
```

Thus this would break.

Reply:
We could default the status to `new` is the order is not yet saved in DB.

Reply:
The second arg is `$key`, this would actually return `null` as `$previous_status`. But what you say makes sense, an order that exists only in memory going from null status into one of the triggering status should trigger the new order hook!

Changed in https://github.com/woocommerce/woocommerce/pull/49098/commits/f0b4303eea5c00f9e47cf28acc54af531c40fb6f